added option to fix torsional angle

vtlim opened the pull request.

vtlim committed:
Merge pull request #1 from magee256/master

autodefine dsp functionality, remove directory name check of params

vtlim committed:
added option to fix torsion

magee256 commented on SetUp/autoDefine/autoDefine.py:
```diff
@@ -163,7 +163,7 @@ def optsParser( opts ):
 #                       for one. 
 #       save_intermed - If true do not remove files generated in setting up control
-def inputBuilder( directories, options, keep_going, save_intermed ): 
+def inputBuilder( directories, options, keep_going, save_intermed, cosmo ): 
```
Cosmo more logically belongs in the options file, better implementation would be to use a command writer that returns True if $cosmo field is present

magee256 commented on SetUp/autoDefine/commandWriters.py:
```diff
@@ -193,6 +193,35 @@ def assignSym(botSpecs, entries, defInp):
+#
+#Default: $fix omitted - restraint not used
+#
+#KeyFormat: $fix type=[label] atoms=[label1,label2,label3,label4]
```
To get multiple specifications a better format would be [type]=[1-4,6]. This allows easy support for all specification types

vtlim replied:
I don't know what you mean by [1-4,6].

magee256 replied:
That's my shorthand for a list of integers, it's meant to show ranges and comma separated lists are allowed. Maybe not that clear without context.

magee256 commented on SetUp/autoDefine/commandWriters.py:
```diff
@@ -193,6 +193,35 @@ def assignSym(botSpecs, entries, defInp):
+   fix=getLine(entries, key)
+   fixBot=getLine(botSpecs, key)
+
+   if fixBot:
```
Should put a warning here if $fix was already defined, seems likely people would want to tack more frozen coords on.

vtlim replied:
Again, can you clarify?

magee256 replied:
I can do this. I basically just mean insert the lines: 
```
if fix:
     print('Warning: frozen coordinates specified in top level options file will be overridden')
```

After `if fixBot:`

magee256 commented on SetUp/autoDefine/autoDefine.py:
```diff
@@ -277,16 +278,19 @@ def inputBuilder( directories, options, keep_going, save_intermed ):
-#        for i in ['input.xyz','options','templateCosmo.input','cosmoprep.out','def.input','def.out']:
-#            try: os.rename(i,'setup/'+i)
-#            except OSError: pass
+        if cosmo:
```
This is better than before, but I still don't want this in the code until cosmoprep.input is generated by the script

magee256 commented on SetUp/autoDefine/autoDefine.py:
```diff
@@ -321,7 +325,10 @@ def inputBuilder( directories, options, keep_going, save_intermed ):
             help='Continue iterating through directories even if define fails.')
     parser.add_argument('-i','--save_intermed', action='store_true',
             help='Do not remove files generated while setting up control')
+    parser.add_argument('-c','--cosmo', default=False, action='store_true',
```
Don't make the user specify this on the command line, better to centralize calculation specifications

magee256 commented on SetUp/autoDefine/commandWriters.py:
```diff
@@ -756,6 +785,7 @@ def scf(botSpecs, entries, defInp):
 
       defInp.write('\n')
 
+
```
?

magee256 reviewed (changes requested):
A good starting point for a $fix menu, but generalizing to more options is simple and should be done before accepting pull. Also try to keep command line options pertinent to the set up script, not the calculation being set up

vtlim commented:
I'll work on the cosmo stuff, but if you want to generalize fix, you can do that part.

vtlim committed:
incorporated cosmo input writer

vtlim commented:
@magee256, cosmo sections have been updated.

vtlim committed:
fix file copying for cosmoprep.input

magee256 committed:
Modify cosmo section of autoDefine.py

Removed the lines of code transferring files to a setup directory, instead input and output generated by the script are removed if save_intermed is False.

magee256 committed:
Modify cosmo command writer

magee256 committed:
Generalize fix command writer

Modified constraint specification format, allowing for multiple specifications and more accepted specification types. Also included error checking.
